### PR TITLE
Added tripperwares for injecting backend & proxy auth to winch based on route config.

### PR DESCRIPTION
**Pull request by Bplotka**
Please review @kdima

`Signed-off-by:` Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: lib/auth/iface.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package auth
```
Comment (Bplotka) [inline lib/auth/iface.go]: If not then where I should put it?

File: lib/http/tripperware/backend_auth.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	authSource := route.BackendAuth
+	if authSource == nil {
+		// No auth configured.
+		return t.parent.RoundTrip(req)
+	}
+
+	tags := http_ctxtags.ExtractInbound(req)
+	tags.Set("http.auth", authSource.Name())
```
Comment (Bplotka) [inline lib/http/tripperware/backend_auth.go]: +1

File: lib/http/tripperware/backend_auth.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package tripperware
```
Comment (Bplotka) [inline lib/http/tripperware/backend_auth.go]: you mean unit test for these?

File: proto/winch/config/mapper.proto [hunk 4]
```diff
@@ -5,22 +5,25 @@ package winch.config;
...
 
-/// Config is the top level configuration message for a winch mapper.
+/// MapperConfig is the top level configuration message for a winch mapper.
 message MapperConfig {
     repeated Route routes = 1;
 }
 
 message Route {
+    // Optional auth injection. Reference to AuthSource.
+    string backend_auth = 1;
```
Comment (Bplotka) [inline proto/winch/config/mapper.proto]: Discussed offlined

**Comment by Bplotka** [comment]
codecov is confused - last master commit does not have coverage

File: lib/http/tripperware/backend_auth.go (hunk 3 above)
Comment (kdima) [inline lib/http/tripperware/backend_auth.go]: yeap

File: lib/http/tripperware/map.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	parent http.RoundTripper
+}
+
+func requestWithRoute(req *http.Request, r *kedge_map.Route) *http.Request {
+	return req.WithContext(context.WithValue(req.Context(), routeContextKey, r))
+}
+
+func (t *mappingTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	host := req.URL.Host
+	if strings.Contains(host, ":") {
```
Comment (kdima) [inline lib/http/tripperware/map.go]: this is reused in a bunch of places. Factor it out inot a function

File: lib/http/ctxtags/tags.go [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package ctxtags
+
+const (
```
Comment (kdima) [inline lib/http/ctxtags/tags.go]: commnet here describing usage

Comment (Bplotka) [inline lib/http/ctxtags/tags.go]: done

**Pull request merged**